src/map/pc_groups.cpp

```diff
@@ -3,462 +3,356 @@
-				ShowConfigWarning(group, "pc_groups:read_config: duplicate group name %s, removing...", groupname);
-				config_setting_remove_elem(groups, i);
-				--i;
-				--group_count;
-				continue;
+			bool allowed = it.second.as<bool>();
+
+			group->commands[command] = allowed;
+		}
+	}
+
+	if( this->nodeExists( node, "CharCommands" ) ){
+		const YAML::Node& commands = node["CharCommands"];
+
+		for( const auto& it : commands ){
+			std::string command = it.first.as<std::string>();
+
+			util::tolower( command );
+
+			if( !atcommand_exists( command.c_str() ) ){
+				this->invalidWarning( it, "Unknown atcommand: %s\n", command.c_str() );
+				return 0;
 			}
 
-			CREATE(group_settings, GroupSettings, 1);
-			group_settings->id = id;
-			group_settings->level = level;
-			group_settings->name = groupname;
-			group_settings->log_commands = log_commands != 0;
-			group_settings->inherit = config_setting_get_member(group, "inherit");
-			group_settings->commands = config_setting_get_member(group, "commands");
-			group_settings->permissions = config_setting_get_member(group, "permissions");
-			group_settings->inheritance_done = false;
-			group_settings->root = group;
-			group_settings->group_pos = i;
-
-			strdb_put(pc_groupname_db, groupname, group_settings);
-			idb_put(pc_group_db, id, group_settings);
-			
+			bool allowed = it.second.as<bool>();
+
+			group->char_commands[command] = allowed;
 		}
-		group_count = config_setting_length(groups); // Save number of groups
-		
-		// Check if all commands and permissions exist
-		iter = db_iterator(pc_group_db);
-		for (group_settings = (GroupSettings *)dbi_first(iter); dbi_exists(iter); group_settings = (GroupSettings *)dbi_next(iter)) {
-			config_setting_t *commands = group_settings->commands, *permissions = group_settings->permissions;
-			int count = 0, j;
-
-			// Make sure there is "commands" group
-			if (commands == NULL)
-				commands = group_settings->commands = config_setting_add(group_settings->root, "commands", CONFIG_TYPE_GROUP);
-			count = config_setting_length(commands);
-
-			for (j = 0; j < count; ++j) {
-				config_setting_t *command = config_setting_get_elem(commands, j);
-				const char *name = config_setting_name(command);
-				if (!atcommand_exists(name)) {
-					ShowConfigWarning(command, "pc_groups:read_config: non-existent command name '%s', removing...", name);
-					config_setting_remove(commands, name);
-					--j;
-					--count;
+	}
+
+	if( this->nodeExists( node, "Permissions" ) ){
+		const YAML::Node& permissions = node["Permissions"];
+
+		for( const auto& it : permissions ){
+			std::string permission = it.first.as<std::string>();
+			bool allowed = it.second.as<bool>();
+			const char* str = permission.c_str();
+			bool found = false;
+
+			for( const auto& permission_name : pc_g_permission_name ){
+				if( strcmpi( permission_name.name, str ) == 0 ){
+					if( allowed ){
+						group->permissions |= permission_name.permission;
```

Comment:
Maybe change the permissions to bitset so we won't be over indulged by large bitmasks again.